nowcasting_dataset/geospatial.py

```diff
@@ -73,3 +76,28 @@ def lat_lon_to_osgb(lat: Number, lon: Number) -> Tuple[Number, Number]:
       lat, lon: Location is WGS84 coordinates.
     """
     return transformers.lat_lon_to_osgb.transform(lat, lon)
+
+
+def calculate_azimuth_and_elevation_angle(location_x: float, location_y: float, datestamps: [datetime]) -> pd.DataFrame:
+    """
+    Calculation the azimuth angle, and the elevation angle for several datetamps, but for one specific osgb location
+
+    More details see: https://www.celestis.com/resources/faq/what-are-the-azimuth-and-elevation-of-a-satellite/
+
+    Args:
+        location_x: osgb x coordinate
+        location_y: osgb y coordinate
+        datestamps: list of datestamps to calculate the sun angles. i.e the sun moves from east to west in the day.
+
+    Returns: Pandas data frame with the index the same as 'datestamps', with columns of "elevation" and "azimuth" that
+    have been calculate.
+
+    """
+    # change osgb to lat and lon
+    lat, lon = osgb_to_lat_lon(location_x, location_y)
```

Comment:
A tiny tiny suggestion:

Instead of passing OSGB coordinates into `calculate_azimuth_and_elevation_angle` and then converting to lat, lon; instead it would be a tiny bit simpler to take the `latitude` and `longitude` columns from the `pv_metadata` DataFrame.  Then  `calculate_azimuth_and_elevation_angle` wouldn't have to convert to lat, lon.  But it's no biggie!

Reply:
Yea, thats a good idea, that'll save a tiny bit of work, so why not

Reply:
done